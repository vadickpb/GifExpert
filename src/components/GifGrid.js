import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {


    const { data:images, loading} = useFetchGif(category);

    return (

        <>
        
            <h3 className= "animate__animated animate__fadeIn">{ category }</h3>

            <p>{ loading && 'Cargando..' }</p>
            <div className="card-grid">
                
                    
                    {
                        images.map(img => 
                        <GifGridItem key = { img.id} {...img}/>
                        )
                    }

                    
                
            </div>
        </>
    )
}
