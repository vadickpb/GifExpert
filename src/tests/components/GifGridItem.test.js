import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas para el componente >GifGridItem', () => {
    
    const title = 'titulo'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title = {title} url = {url}/>);

    test('debe mostrar el wrapper', () => { 

        expect(wrapper).toMatchSnapshot();

    });

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title)
        
    })

    test('debe tener una imagen los los atributos src el url y alt con el title', () => {
        const img = wrapper.find('img');
        
        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );
    })

    test('debe de tener un div con la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        console.log(div.hasClass('animate__fadeIn'));
        expect(div.hasClass('animate__fadeIn')).toBe(true)
    })
    
    
    
    
})
