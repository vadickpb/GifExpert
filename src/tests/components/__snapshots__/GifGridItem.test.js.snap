// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente >GifGridItem debe mostrar el wrapper 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
